src/examples/sedov_solution/main.cpp

```diff
@@ -0,0 +1,199 @@
+    // auto         constants = sedovConstants();
+    // const size_t dim       = constants["dim"];
+    // const double r0        = constants["r0"];
+    // const double r1        = constants["r1"];
+    // const double eblast    = constants["energyTotal"];
+    // const double gamma     = constants["gamma"];
+    // const double omega     = constants["omega"];
+    // const double rho0      = constants["rho0"];
+    // const double u0        = constants["u0"];
+    // const double p0        = constants["p0"];
+    // const double vr0       = constants["vr0"];
+    // const double cs0       = constants["cs0"];
+
+    const size_t dim = 3;
+    const double r0 = 0.0;
+    const double r1 = 0.5;
+    const double eblast = 1.0;
+    const double gamma = 5.0 / 3.0;
+    const double omega = 0.0;
+    const double rho0 = 1.0;
+    const double u0 = 1e-8;
+    const double p0 = 0.0;
+    const double vr0 = 0.0;
+    const double cs0 = 0.0;
+
+    double shockFront;
+    {
+        std::vector<double> rDummy(1, 0.1);
+        std::vector<Real> rho(1), p(1), u(1), vel(1), cs(1);
+        shockFront = SedovComputer::sedovSol(dim, time, eblast, omega, gamma, rho0, u0, p0, vr0, cs0, rDummy, rho, p, u,
+                                             vel, cs);
+    }
+
+    // Set the positions for calculating the solution
+    size_t nSteps = 100000;
+    size_t nSamples = nSteps + 2;
+    std::vector<double> rSol(nSamples);
+
+    const double rMax = 2. * r1;
+    const double rStep = (rMax - r0) / nSteps;
+
+    for (size_t i = 0; i < nSteps; i++) {
+        rSol[i] = (r0 + (0.5 * rStep) + (i * rStep));
+    }
+    rSol[nSamples - 2] = shockFront;
+    rSol[nSamples - 1] = shockFront + 1e-7;
+    std::sort(std::begin(rSol), std::end(rSol));
+
+    // analytical solution output
+    std::vector<Real> rho(nSamples), p(nSamples), u(nSamples), vel(nSamples), cs(nSamples);
+
+    // Calculate theoretical solution
+    SedovComputer::sedovSol(dim, time, eblast, omega, gamma, rho0, u0, p0, vr0, cs0, rSol, rho, p, u, vel, cs);
+
+    if (normalize) {
+        std::for_each(std::begin(rho), std::end(rho), [](auto& val) { val /= SedovComputer::rho_shock; });
+        std::for_each(std::begin(u), std::end(u), [](auto& val) { val /= SedovComputer::u_shock; });
+        std::for_each(std::begin(p), std::end(p), [](auto& val) { val /= SedovComputer::p_shock; });
+        std::for_each(std::begin(vel), std::end(vel), [](auto& val) { val /= SedovComputer::vel_shock; });
+        std::for_each(std::begin(cs), std::end(cs), [](auto& val) { val /= SedovComputer::cs_shock; });
+    }
+
+    writeColumns1D(solFile);
+    writeAscii<Real>(0, nSteps, solFile, true, {rSol.data(), rho.data(), u.data(), p.data(), vel.data(), cs.data()},
+                     std::setw(16), std::setprecision(7), std::scientific);
+
+    std::cout << "Created solution file: '" << solFile << std::endl;
+
+    return EXIT_SUCCESS;
+}
+
+void printHelp(char* binName) {
+    printf("\nUsage:\n\n");
```

Comment:
No idea why the original author used `printf` instead of `std::cout`

Reply:
Yeah not sure. Tried to not modify most of that code though.